Fix GEOSException when reprojecting antimeridian-spanning geometries

egagli opened the pull request:
Fixes GEOSException when reprojecting data in sinusoidal projection that spans the antimeridian

**`odc/geo/geobox.py`:**
- Enhanced `GeoboxTiles.grid_intersect()` with error handling for antimeridian-spanning geometries
- Added `wrapdateline` parameter to `GeoBox.footprint()` method for antimeridian-aware geometry conversion

**`tests/antimeridian.py`:**
-Added antimeridian scenario

I tried the scenario in issue #147 and it seems to work? But couldn't figure out how to make it into a test, or if that's even desirable.

egagli committed:
Fix GEOSException when reprojecting antimeridian-spanning geometries. Add wrapdateline argument, GeoBoXTiles.grid_intersect error handing, and a test. Hopefully addresses issue #147.

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

SpacemanPaul reviewed (changes requested):
Thanks, this looks useful.  

If you can address the issues picked up by pylint, black and mypy, I'd be happy to merge.

egagli committed:
Fix antimeridian geometry handling and code quality issues

- Added robust exception handling for GEOSException when reprojecting antimeridian-spanning geometries
- Fixed mypy type safety issues (CRS null checks, proper typing)
- Fixed pylint warnings (unused variables, added justified disable comments)
- Fixed black formatting issues (split long lines, reformatted function signatures)
- All tests passing (pytest tests/test_antimeridian.py and tests/test_geobox.py)
- Code quality checks passing (pylint 10.00/10, mypy clean, black formatted)

Ready for merge.
